Index: setup.py
===================================================================
--- setup.py
+++ setup.py	2011-06-28 13:57:57.616850008 +0200
@@ -54,6 +54,9 @@
 """ % sonata.svnversion.VERSION)
 versionfile.close()
 
+extra_link_args = capture("pkg-config --libs gtk+-2.0 pygtk-2.0").split()
+extra_link_args.append("-lX11")
+
 setup(name='Sonata',
         version="1.6.2.1",
         description='GTK+ client for the Music Player Daemon (MPD).',
@@ -74,7 +77,7 @@
         ext_modules=[Extension(
         "mmkeys", ["mmkeys/mmkeyspy.c", "mmkeys/mmkeys.c", "mmkeys/mmkeysmodule.c"],
         extra_compile_args=capture("pkg-config --cflags gtk+-2.0 pygtk-2.0").split(),
-        extra_link_args=capture("pkg-config --libs gtk+-2.0 pygtk-2.0").split()
+        extra_link_args=extra_link_args, 
          ),],
         scripts = ['sonata/sonata'],
         data_files=[('share/sonata', ['README', 'CHANGELOG', 'TODO', 'TRANSLATORS']),
